lib/timespan.py

```diff
@@ -0,0 +1,89 @@
+import re
```

Comment:
Looks like a good first implementation. I'd have created a base abstract class and then moved the `parse_timespan` logic to the ETL classes.

These time spans are pretty common but in the future, we could end up having some tightly coupled ones but that's a "future"